src/joint_moteus.cpp

```diff
@@ -0,0 +1,294 @@
+/**
+ * @file joint_moteus.cpp
+ * @author J. Diego Caporale (jdcap@seas.upenn.edu), Lucien Peach 
+ * (peach@seas.upenn.edu), Shane Rozen-Levy (srozen01@seas.upenn.edu)
+ * @brief Moteus powered joint class implementation
+ * @date 2023-07-13
+ * 
+ * @copyright BSD 3-Clause License, Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+ * 
+ */
+
+
+#include "kodlab_mjbots_sdk/joint_moteus.h"
+
+namespace kodlab{
+namespace mjbots{
+
+// kDefaultQuery defintion
+const ::mjbots::moteus::QueryCommand JointMoteus::kDefaultQuery{
+  ::mjbots::moteus::Resolution::kInt8,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore
+};
+
+// kTorqueQuery definition
+const ::mjbots::moteus::QueryCommand JointMoteus::kTorqueQuery{
+  ::mjbots::moteus::Resolution::kInt8,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt32,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore
+};
+
+// kDebugQuery definition
+const ::mjbots::moteus::QueryCommand JointMoteus::kDebugQuery{
+  ::mjbots::moteus::Resolution::kInt8,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt32,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kIgnore,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt8
+};
+
+// kComprehensiveQuery definition
+const ::mjbots::moteus::QueryCommand JointMoteus::kComprehensiveQuery{
+  ::mjbots::moteus::Resolution::kInt8,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt32,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kInt16,
+  ::mjbots::moteus::Resolution::kIgnore,
```

Comment:
what is kIgnore in comprehensive? That doesn't make sense.

Reply:
The value associated with this kIgnore is "rezero_state", for which there is no documentation within the mjbots `reference.md` file. As such, I opted to set the resolution to kIgnore, as I was unsure what information this contains / how it would interact with the closed-loop control (if at all). 

Think that it would be better to include all fields, like suggested, but either want to know what this field does or perhaps change the name of the query.

Reply:
I think thats fine, just add a comment.